Utils/azuretools-core/src/com/microsoft/azuretools/core/mvp/model/mysql/MySQLMvpModel.java

```diff
@@ -229,60 +229,41 @@ public static boolean enableAllowAccessFromLocalMachine(final String subscriptio
             if (MySQLMvpModel.FirewallRuleMvpModel.isAllowAccessFromLocalMachine(subscriptionId, server)) {
                 return true;
             }
-            try {
-                final String publicIp = MySQLMvpModel.FirewallRuleMvpModel.getPublicIp();
-                final String ruleName = NAME_PREFIX_ALLOW_ACCESS_TO_LOCAL + publicIp.replaceAll("\\.", "-");
-                final FirewallRuleInner firewallRule = new FirewallRuleInner();
-                firewallRule.withStartIpAddress(publicIp);
-                firewallRule.withEndIpAddress(publicIp);
-                final MySQLManager mySQLManager = AuthMethodManager.getInstance().getMySQLManager(subscriptionId);
-                mySQLManager.firewallRules().inner().createOrUpdate(server.resourceGroupName(), server.name(), ruleName, firewallRule);
-            } catch (IOException e) {
-                Lombok.sneakyThrow(e);
+            MySQLConnectionUtils.ConnectResult connectResult = MySQLConnectionUtils.connectWithPing(JdbcUrl.mysql(server.fullyQualifiedDomainName()),
+                    server.administratorLogin() + "@" + server.name(), StringUtils.EMPTY);
```

Comment:
this will certainly fail since password is not specified. won't it?

Reply:
these lines only care if the message contains IP or not.
authenication is verified only if IP is set in firewall rules.